### PR TITLE
8302158: PPC: test/jdk/jdk/internal/vm/Continuation/Fuzz.java: AssertionError: res: false shouldPin: false

### PR DESCRIPTION
**Pull request by reinrich**
This fixes the linked issue by trimming the caller of a frame to be deoptimized back to its `unextended_sp` iff it is compiled. The creation of the section `dead after deoptimization` shown in the attachment [yield_after_deopt_failure.log](https://bugs.openjdk.org/secure/attachment/102602/yield_after_deopt_failure.log) is prevented by this.

A new mode is added to the test BasicExt.java where all frames are deoptimized after a yield operation. The issue can be deterministically reproduced with the new mode. It's not worth to execute all test cases with the new mode though. Instead `ContinuationCompiledFramesWithStackArgs_3c4` is always executed a 2nd time in this mode.

Before this BasicExt.java was refactored for better argument processing and representation of the test modes.
Also the try-catch-clause in the main method had to be changed to rethrow the caught exception because without this the test would have succeeded.

Testing: jtreg tests tier 1-4 on standard platforms and also on ppc64le.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8302158](https://bugs.openjdk.org/browse/JDK-8302158): PPC: test/jdk/jdk/internal/vm/Continuation/Fuzz.java: AssertionError: res: false shouldPin: false

### Reviewers
 * [Goetz Lindenmaier](https://openjdk.org/census#goetz) (@GoeLin - **Reviewer**) ⚠️ Review applies to [7ca17977](https://git.openjdk.org/jdk/pull/12557/files/7ca179771a663663fa4e0693ff0b1fe8336fff3d)
 * [Martin Doerr](https://openjdk.org/census#mdoerr) (@TheRealMDoerr - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12557/head:pull/12557` \
`$ git checkout pull/12557`

Update a local copy of the PR: \
`$ git checkout pull/12557` \
`$ git pull https://git.openjdk.org/jdk pull/12557/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12557`

View PR using the GUI difftool: \
`$ git pr show -t 12557`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12557.diff">https://git.openjdk.org/jdk/pull/12557.diff</a>

</details>

### CODE REVIEW

**Review by GoeLin — APPROVED** [review]
LGTM

File: src/hotspot/cpu/ppc/frame_ppc.cpp [hunk 1]
```diff
@@ -421,8 +421,12 @@ void frame::describe_pd(FrameValues& values, int frame_no) {
 #endif
 
 intptr_t *frame::initial_deoptimization_info() {
-  // unused... but returns fp() to minimize changes introduced by 7087445
-  return fp();
+  // `this` is the caller of the deoptee. We want to trimm it, if compiled, to
```
Comment (TheRealMDoerr) [inline src/hotspot/cpu/ppc/frame_ppc.cpp:424]: I believe "to trim" should contain only one 'm' in English.

**Review by TheRealMDoerr — APPROVED** [review]
LGTM. Thanks for fixing!

File: src/hotspot/cpu/ppc/sharedRuntime_ppc.cpp [hunk 2]
```diff
@@ -3090,6 +3090,11 @@ void SharedRuntime::generate_deopt_blob() {
 
   // stack: (caller_of_deoptee, ...).
 
+  // Freezing continuation frames requires that the caller is trimmed to unextended sp if compiled.
+  Register caller_sp = R23_tmp3;
```
[the remark this answers is not in the document]
Comment (reinrich) [reply src/hotspot/cpu/ppc/sharedRuntime_ppc.cpp:3096]: Yes it can be interpreted. I've extended the comment for this case.

**Review by TheRealMDoerr — APPROVED** [review]
Thanks!

**Comment by reinrich** [comment]
Thanks for the reviews!

/integrate

**Pull request closed without merging**